renormalizer/mps/mps.py

```diff
@@ -1495,50 +1495,26 @@ def full_wfn(self) -> np.array:
     def _calc_reduced_density_matrix(self, mp1, mp2):
         if isinstance(self.mol_list, MolList2):
             assert mp1 is None and mp2 is None
-            if not self.mol_list.multi_electron:
-                reduced_density_matrix = np.zeros(
-                    (self.mol_list.n_edofs, self.mol_list.n_edofs),
-                    dtype=backend.complex_dtype)
-                for idx in range(self.mol_list.n_edofs):
-                    for jdx in range(idx, self.mol_list.n_edofs):
-                        model = {(f"e_{idx}", f"e_{jdx}"):[(Op(r"a^\dagger", 1),
-                            Op("a",-1), 1.0)]}
+            key = "reduced_density_matrix"
+            n_e = self.mol_list.n_edofs
+            if key not in self.mol_list.mpos:
+                mpos = []
+                for idx in range(n_e):
+                    for jdx in range(idx, n_e):
+                        model = {(f"e_{idx}", f"e_{jdx}"): [(Op(r"a^\dagger", 1),
+                                                             Op("a", -1), 1.0)]}
                         mpo = Mpo.general_mpo(self.mol_list, model=model,
-                                model_translator=ModelTranslator.general_model)
-                        reduced_density_matrix[idx, jdx] = self.expectation(mpo)
-                        reduced_density_matrix[jdx, idx] = np.conj(reduced_density_matrix[idx, jdx])
+                                              model_translator=ModelTranslator.general_model)
+                        mpos.append(mpo)
+                self.mol_list.mpos[key] = mpos
             else:
-                e_idx = self.mol_list.multi_e_idx
-                l = ones((1, 1))
-                for i in range(e_idx):
-                    mat = self[i].reshape(self[i].shape[0], -1, self[i].shape[-1])
-                    l = tensordot(l, mat, axes=(0, 0))
-                    l = tensordot(l, mat.conj(), axes=([0, 1], [0, 1]))
-                r = ones((1, 1))
-                for i in range(len(self)-1, e_idx, -1):
-                    mat = self[i].reshape(self[i].shape[0], -1, self[i].shape[-1])
-                    r = tensordot(mat, r, axes=(-1, 0))
-                    r = tensordot(r, mat.conj(), axes=([1, 2], [1, 2]))
```

Comment:
If this part is deleted,  the `BasisMultiElectron` class can not calculate the reduced density matrix anymore.